source/components/newevent.jsx

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from 'react'
+import { Input, Button, Modal, Form, Select, Message, Alert } from 'semantic-ui-react'
+import Notifications, {notify} from 'react-notify-toast';
+
+// import styles from './newevent.scss'
+
+import axios from 'axios'
+
+class NewEvent extends Component {
+
+    constructor() {
+        super();
+        this.state = {
+            name: '',
+            category: '',
+            points: 0,
+            date: '',
+            startTime: '',
+            endTime: '',
+            password: '',
+            showMsg: false,
+            message: '',
+        }
+    }
+
+    handleChange = (event, data) => {
+        console.log(`${event.target.id}: ${data.value}`)
+        this.setState({
+            [event.target.id]: data.value
+        })
+    }
+
+    // fixes off by one date error
+    formatDate = date => {
+        const newDate = new Date(date)
+        newDate.setDate(newDate.getDate() + 1)
+        return newDate
+    }
+
+    createEvent = async () => {
+        const date = this.formatDate(this.state.date)
+        const event = {
+            name: this.state.name,
+            category: this.state.category,
+            points: this.state.points,
+            date,
+            startTime: this.state.startTime,
+            endTime: this.state.endTime,
+            password: this.state.password,
+        }
+        const res = await axios.post('http://localhost:3000/api/events', event)
+        console.log(res)
+        if (res.data.code === 500) {
```

Comment:
prob better for the user if we validate on frontend instead of backend for these
password/server error can be backend